### PR TITLE
Added searching for chefignore file in cookbook root directory

### PR DESCRIPTION
**Pull request by LukeRoz**
Signed-off-by: Lukasz Rozok <lukescroz@gmail.com>

### Description

Fixes reading chefignore file from cookbook root directory.

### Issues Resolved

https://github.com/chef/chef/issues/2994

### Check List

- [ ] New functionality includes tests
- [ ] All tests pass
- [ ] RELEASE\_NOTES.md, has been updated if required (not required for bugfixes, required for API changes)
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Comment by lamont-granquist** [comment]
For reference this is what berkshelf has always done:

https://github.com/berkshelf/berkshelf/blob/master/lib/berkshelf/berksfile.rb#L603-L620

**Comment by thommay** [comment]
Hey @LukeRoz this looks great; could you fix up chefstyle (just run `chefstyle -a`) and rebase and we'll get this merged.
A potential optimization might be to use `Pathname.ascend`, but that would be a nice to have rather than a blocker.

**Comment by LukeRoz** [comment]
Hi @thommay,

Sorry for late reply. One of offenses in my code is related to using "for" instead of "each" ([chefignore.rb#L64](https://github.com/LukeRoz/chef/blob/22ed661e9099343db5cee4a60a88be527a637d21/lib/chef/cookbook/chefignore.rb#L64)). To be honest "each" doesn't fit in here for me. Any ideas how could I change it?

File: lib/chef/cookbook/chefignore.rb [hunk 1]
```diff
@@ -56,22 +54,32 @@ def parse_ignore_file
...
         ignore_globs
       end
 
       def find_ignore_file(path)
-        if File.basename(path) =~ /chefignore/
-          path
-        else
-          File.join(path, "chefignore")
+        ignore_path=path
+        for i in 0..2
```
Comment (coderanger) [inline lib/chef/cookbook/chefignore.rb]: `(0..2).each do |i|` or `3.times do |i|`

File: lib/chef/cookbook/chefignore.rb [hunk 2]
```diff
@@ -56,22 +54,32 @@ def parse_ignore_file
             ignore_globs << line.strip unless line =~ COMMENTS_AND_WHITESPACE
           end
         else
-          Chef::Log.trace("No chefignore file found at #{@ignore_file} no files will be ignored")
+          Chef::Log.debug("No chefignore file found. No files will be ignored!")
```
Comment (coderanger) [inline lib/chef/cookbook/chefignore.rb]: This needs to be updated.

**Comment by coderanger** [comment]
Do we not have helpers for this kind of thing already? This looks like some very intricate code that we probably shouldn't need to be so intricate :)

File: lib/chef/cookbook/chefignore.rb (hunk 2 above)
Comment (thommay) [reply lib/chef/cookbook/chefignore.rb]: I think this is actually correct - a `debug` level message seems reasonable.

**Comment by lamont-granquist** [comment]
@coderanger no, we don't have helpers, and berkshelf does its own different thing which is why I couldn't reuse code there.

Hence my suggestion to standardize on what berkshelf does since it seems it has been doing that for a long time.

**Comment by lamont-granquist** [comment]
OTOH, this is better than what we've got now though....

**Comment by lamont-granquist** [comment]
looks like there may still be a chefstyle failure tho...

**Comment by LukeRoz** [comment]
Son of a... OK, double-quoted strings are required. I'll update it tomorrow morning :)

**Comment by tas50** [comment]
Closing this in favor of #8925 which solves this problem plus several others

**Pull request closed without merging**